spec/operators/combineAll-spec.ts

```diff
@@ -1,453 +1,522 @@
+      const expected = '|';
+
+      const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with hot-empty and hot-single', () => {
-    const e1 =        hot('-a-^-|');
-    const e1subs =           '^ !';
-    const e2 =        hot('-b-^-c-|');
-    const e2subs =           '^   !';
-    const expected =         '----|';
-
-    const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const e1 = hot('-a-^-|');
+      const e1subs = '   ^-!';
+      const e2 = hot('-b-^-c-|');
+      const e2subs = '   ^---!';
+      const expected = ' ----|';
+
+      const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with hot-single and hot-empty', () => {
-    const e1 =        hot('-a-^-|');
-    const e1subs =           '^ !';
-    const e2 =        hot('-b-^-c-|');
-    const e2subs =           '^   !';
-    const expected =         '----|';
-
-    const result = of(e2, e1).pipe(combineAll((x, y) => x + y));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const e1 = hot('-a-^-|');
+      const e1subs = '   ^-!';
+      const e2 = hot('-b-^-c-|');
+      const e2subs = '   ^---!';
+      const expected = ' ----|';
+
+      const result = of(e2, e1).pipe(combineAll((x, y) => x + y));
+
+      expectObservable(result).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with hot-single and never', () => {
-    const e1 =        hot('-a-^-|');
-    const e1subs =           '^ !';
-    const e2 =        hot('------'); //never
-    const e2subs =           '^  ';
-    const expected =         '-'; //never
-
-    const result = of(e1, e2).pipe(combineAll((x, y) => x + y));
-
-    expectObservable(result).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const e1 = hot('-a-^-|');
+      const e1subs = '   ^-!';
+      const e2 = hot('------'); //never
+      const e2subs = '   ^--';
+      const expected = ' -'; //never
```

Comment:
If you want, you can extend the `-` in `expected` to line up with the end of `e2` - like in the test that follows this one. If the marble diagram has nothing but `-` in it, where is ends doesn't matter.